lib/jekyll-remote-theme/downloader.rb

```diff
@@ -38,16 +36,40 @@ def downloaded?
       private
 
       def zip_file
-        @zip_file ||= Tempfile.new([TEMP_PREFIX, ".zip"])
+        @zip_file ||= Tempfile.new([TEMP_PREFIX, ".zip"], :binmode => true)
       end
 
       def download
         Jekyll.logger.debug LOG_KEY, "Downloading #{zip_url} to #{zip_file.path}"
-        request = Typhoeus::Request.new zip_url, TYPHOEUS_OPTIONS
-        request.on_headers  { |response| raise_if_unsuccessful(response) }
-        request.on_body     { |chunk| zip_file.write(chunk) }
-        request.on_complete { |response| raise_if_unsuccessful(response) }
-        request.run
+        Net::HTTP.start(zip_url.host, zip_url.port, :use_ssl => true) do |http|
+          http.request(request) do |response|
+            raise_unless_sucess(response)
+            enforce_max_file_size(response.content_length)
+            response.read_body do |chunk|
+              zip_file.write chunk
+            end
+          end
+        end
+        @downloaded = true
+      rescue *NET_HTTP_ERRORS => e
+        raise DownloadError, e.message
+      end
+
+      def request
+        return @request if defined? @request
+        @request = Net::HTTP::Get.new zip_url.request_uri
+        @request["User-Agent"] = USER_AGENT
+        @request
+      end
+
+      def raise_unless_sucess(response)
+        return if response.is_a?(Net::HTTPSuccess)
+        raise DownloadError, "#{response.code} - #{response.message}"
```

Comment:
Would it be helpful to include the URL that failed to return a helpful response?

Reply:
It would theoretically be in the log output immediately above.